test/box-luatest/downgrade_test.lua

```diff
@@ -0,0 +1,799 @@
+--
+--  tarantool> CREATE TABLE t (s INTEGER PRIMARY KEY, CHECK(LEAST(s, 10) = 10));
+--  ---
+--  - null
+--  - 'Failed to create check constraint ''ck_unnamed_T_1'': Function ''LEAST''
+--    does not exist'
+--  ...
+g.test_downgrade_insert_back_sql_builtin_functions = function(cg)
+    cg.server:exec(function()
+        local fun = require('fun')
+        local helper = require('test.box-luatest.downgrade_helper')
+
+        box.space._func:create_index('__language',
+                                     {parts = {{5, 'string'}}, unique = false})
+        local rows = box.space._func.index.__language:select('SQL_BUILTIN')
+        t.assert_equals(#rows, 0)
+
+        local app_version = helper.app_version('2.9.1')
+        t.assert_equals(box.schema.downgrade_issues(app_version), {})
+        box.schema.downgrade(app_version)
+        rows = box.space._func.index.__language:select('SQL_BUILTIN')
+        t.assert_equals(#rows, 0)
+
+        local prev_version = helper.prev_version('2.9.1')
+        t.assert_equals(box.schema.downgrade_issues(prev_version), {})
+        rows = box.space._func.index.__language:select('SQL_BUILTIN')
+        t.assert_equals(#rows, 0)
+
+        local names_expected = {
+            "TRIM", "TYPEOF", "PRINTF", "UNICODE", "CHAR", "HEX", "VERSION",
+            "QUOTE", "REPLACE", "SUBSTR", "GROUP_CONCAT", "JULIANDAY", "DATE",
+            "TIME", "DATETIME", "STRFTIME", "CURRENT_TIME",
+            "CURRENT_TIMESTAMP", "CURRENT_DATE", "LENGTH", "POSITION", "ROUND",
+            "UPPER", "LOWER", "IFNULL", "RANDOM", "CEIL", "CEILING",
+            "CHARACTER_LENGTH", "CHAR_LENGTH", "FLOOR", "MOD", "OCTET_LENGTH",
+            "ROW_COUNT", "COUNT", "LIKE", "ABS", "EXP", "LN", "POWER", "SQRT",
+            "SUM", "TOTAL", "AVG", "RANDOMBLOB", "NULLIF", "ZEROBLOB", "MIN",
+            "MAX", "COALESCE", "EVERY", "EXISTS", "EXTRACT", "SOME", "GREATER",
+            "LESSER", "SOUNDEX", "LIKELIHOOD", "LIKELY", "UNLIKELY",
+            "_sql_stat_get", "_sql_stat_push", "_sql_stat_init", "GREATEST",
+            "LEAST",
+        }
+
+        box.schema.downgrade(prev_version)
+        rows = box.space._func.index.__language:select('SQL_BUILTIN')
+        local names_actual = fun.totable(fun.map(function(t) return t.name end,
+                                                 rows))
+        t.assert_equals(names_actual, names_expected)
+
+        -- idempotence check
+        box.schema.downgrade(prev_version)
+        rows = box.space._func.index.__language:select('SQL_BUILTIN')
+        local names_actual = fun.totable(fun.map(function(t) return t.name end,
+                                                 rows))
+        t.assert_equals(names_actual, names_expected)
+    end)
+end
+
+------------------------------
+-- Check downgrade from 2.10.0
+------------------------------
+
+-- Manual check:
+--
+--  Tarantool Enterprise 2.11.0-entrypoint-113-g803baaf
```

Comment:
Nit: seems like you copy-pasted 'Tarantool Enterpise' for all tests, even those that are not Enterprise.

Reply:
The manual check was done on enterprise version. Why not?